From 1de3575a1ea9cdb3579ad538108c51818847797c Mon Sep 17 00:00:00 2001
From: Rasmus Thomsen <oss@cogitri.dev>
Date: Fri, 6 Dec 2019 11:48:02 +0100
Subject: [PATCH] revert: autostart-app: Strip blacklisted variables from
 autostart environment

This breaks gnome-session on non-systemd systems.

https://gitlab.gnome.org/GNOME/gnome-session/issues/44
---
 gnome-session/gsm-autostart-app.c | 5 -----
 gnome-session/gsm-util.c          | 6 ------
 gnome-session/gsm-util.h          | 1 -
 3 files changed, 12 deletions(-)

diff --git a/gnome-session/gsm-autostart-app.c b/gnome-session/gsm-autostart-app.c
index 6fd5915..01ba5f9 100644
--- a/gnome-session/gsm-autostart-app.c
+++ b/gnome-session/gsm-autostart-app.c
@@ -997,7 +997,6 @@ autostart_app_start_spawn (GsmAutostartApp *app,
         gboolean         success;
         GError          *local_error;
         const char      *startup_id;
-        const char * const *variable_blacklist;
         const char * const *child_environment;
         int i;
         GAppLaunchContext *ctx;
@@ -1014,10 +1013,6 @@ autostart_app_start_spawn (GsmAutostartApp *app,
         local_error = NULL;
         ctx = g_app_launch_context_new ();
 
-        variable_blacklist = gsm_util_get_variable_blacklist ();
-        for (i = 0; variable_blacklist[i] != NULL; i++)
-                g_app_launch_context_unsetenv (ctx, variable_blacklist[i]);
-
         child_environment = gsm_util_listenv ();
         for (i = 0; child_environment[i] != NULL; i++) {
                 char **environment_tuple;
diff --git a/gnome-session/gsm-util.c b/gnome-session/gsm-util.c
index 02bc4a5..ada1225 100644
--- a/gnome-session/gsm-util.c
+++ b/gnome-session/gsm-util.c
@@ -808,9 +808,3 @@ gsm_util_listenv (void)
         return (const char * const *) child_environment;
 
 }
-
-const char * const *
-gsm_util_get_variable_blacklist (void)
-{
-        return variable_blacklist;
-}
diff --git a/gnome-session/gsm-util.h b/gnome-session/gsm-util.h
index bd7b698..8bca5f4 100644
--- a/gnome-session/gsm-util.h
+++ b/gnome-session/gsm-util.h
@@ -50,7 +50,6 @@ char *      gsm_util_generate_startup_id            (void);
 void        gsm_util_setenv                         (const char *variable,
                                                      const char *value);
 const char * const * gsm_util_listenv               (void);
-const char * const * gsm_util_get_variable_blacklist(void);
 
 gboolean    gsm_util_export_activation_environment  (GError     **error);
 #ifdef HAVE_SYSTEMD
-- 
2.24.0

